### PR TITLE
#149952571 Admin should get a notification for each contribution created

### PR DESCRIPTION
**Pull request by Mnickii**
#### What does this PR do?
-Get notifications for every contribution made, if admin
-Adds a notification endpoint ```/api/v1/notifications```
-Adds a single notification endpoint ```/api/v1/notifications/notification_id```

#### Description of Task to be completed?
-Add a Notification Model
-Add get notifications endpoint
-Add get single notification endpoint
-Add Notification Model tests
-Add notification endpoints tests

#### How should this be manually tested?
At the root of the project directory run the following command:
        
    $ pytest -v 

#### Any background context you want to provide?
For any contribution made, an admin should get a notification. The notification should include which contribution was made.

#### What are the relevant pivotal tracker stories?
#149952571

#### Screenshots (if appropriate)
screen shot 2017-08-23 at 7 06 15 pm
screen shot 2017-08-23 at 7 05 38 pm

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -152,3 +152,21 @@ class Path(db.Model, ModelOpsMixin):
...
+    """ Notification Model. """
+    __tablename__ = "Notification"
+    id = db.Column(db.String(), primary_key=True)
+    user_id = db.Column(db.String, nullable=False)
+    user_name = db.Column(db.String, nullable=False)
+    status = db.Column(db.String(20), default="Unread")
+    created_at = db.Column(db.DateTime, default=datetime.utcnow)
+    contribution_made_id = db.Column('contribution_made_id', db.String,
+                                     db.ForeignKey('Contribution.id'))
+    contribution_archived_id = db.Column('contribution_archived_id', db.String,
```
Comment (Andela-ddiei) [inline api/models.py]: We don't need this yet

File: api/models.py [hunk 2]
```diff
@@ -152,3 +152,21 @@ class Path(db.Model, ModelOpsMixin):
...
+                                     db.ForeignKey('Contribution.id')) #remove
+    contribution_made = db.relationship(
```
Comment (Andela-ddiei) [inline api/models.py]: Remove

File: api/views/contributions.py [hunk 3]
```diff
@@ -136,6 +137,19 @@ def post(self):
...
 
+            generated_notification_id = uuid4().hex
+
+            new_notification = Notification(
+                id=generated_notification_id,
+                user_id=new_contribution.user_id,
+                user_name=user_name,
+                contribution_made_id=new_contribution.id
+            )    
+            db.session.add(new_notification)
```
Comment (Andela-ddiei) [inline api/views/contributions.py]: new_notification.save()

File: api/views/notifications.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    @token_required
+    @admin_required
+    def get_notification(self, notif_id):
+        notification = Notification.query.get(notif_id)
+        if notification:
+            result = notification.serialize()
+            return result
+        else:
+            response = {
```
Comment (Andela-ddiei) [inline api/views/notifications.py]: Make response consistent.

File: api/views/notifications.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+from flask_restful import Resource
+
+from ..models import db, Notification
+from ..auth import token_required, admin_required
+
+class NotificationsResource(Resource):
+
+    @token_required
+    @admin_required
+    def get_notification(self, notif_id):
```
Comment (Andela-ddiei) [inline api/views/notifications.py]: This method doesn't have to be separated from the get method.

File: api/views/paths.py~HEAD [hunk 6]
```diff
@@ -0,0 +1,134 @@
+from flask import request, jsonify
```
Comment (Andela-ddiei) [inline api/views/paths.py~HEAD]: Delete duplicate file.

File: api/views/contributions.py [hunk 7]
```diff
@@ -136,6 +137,19 @@ def post(self):
 
             contribution_serialized = new_contribution.serialize()
 
+            generated_notification_id = uuid4().hex
```
Comment (LarryWachira) [inline api/views/contributions.py]: IDs are autogenerated now

File: api/models.py [hunk 8]
```diff
@@ -152,3 +152,21 @@ class Path(db.Model, ModelOpsMixin):
 
     def __repr__(self):
         return '<Path %r>' % self.name
+
+class Notification(db.Model, ModelOpsMixin):
+    """ Notification Model. """
+    __tablename__ = "Notification"
+    id = db.Column(db.String(), primary_key=True)
```
Comment (LarryWachira) [inline api/models.py]: `db.String` so that it's consistent across this model's columns.

File: api/views/contributions.py [hunk 9]
```diff
@@ -136,6 +137,19 @@ def post(self):
...
+            new_notification = Notification(
+                id=generated_notification_id,
+                user_id=new_contribution.user_id,
+                user_name=user_name,
+                contribution_made_id=new_contribution.id
+            )    
+            db.session.add(new_notification)
+            db.session.commit()
+
+            notification_serialized = new_notification.serialize()
```
Comment (LarryWachira) [inline api/views/contributions.py]: Where is notification_serialized used in this post method?

File: api/views/notifications.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
+from flask import request, url_for, jsonify, g, current_app
+from flask_restful import Resource
+
+from ..models import db, Notification
+from ..auth import token_required, admin_required
+
+class NotificationsResource(Resource):
+
+    @token_required
```
Comment (LarryWachira) [inline api/views/notifications.py]: Since this does not become a route, there's not need use authorization decorators.

File: api/models.py (hunk 8 above)
Comment (Andela-ddiei) [inline api/models.py]: @LarryWachira please colour.

Comment (LarryWachira) [inline api/models.py]: Currently the type being used is `db.string()`. `db.string` would be consistent with all other models if you're not setting a character limit.

**Pull request merged**